src/history/hs_cursor.c

```diff
@@ -102,10 +102,15 @@ __wt_hs_find_upd(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
         goto done;
     }
 
-    WT_ERR_NOTFOUND_OK(__wt_curhs_open(session, NULL, &hs_cursor), true);
-    /* Do this separately for now because the behavior below is confusing if it triggers. */
-    WT_ASSERT(session, ret != WT_NOTFOUND);
-    WT_ERR(ret);
+    if (cbt->hs_cursor != NULL)
+        hs_cursor = cbt->hs_cursor;
+    else {
+        WT_ERR_NOTFOUND_OK(__wt_curhs_open(session, NULL, &hs_cursor), true);
```

Comment:
This change is causing the history store cursor for every data store cursor. Before this change, if the application opens 3 data store cursors that are touching the history store, we opened only 3 cursors, now it is 6 cursors and in that 3 are history store cursors.

I feel this change can cause to open unnecessary extra history store cursors.

Reply:
What is the downside to open more history store cursors except use some extra memory?

Reply:
Yes, this change leads to double the number of cursors, and it can improve the performance of reusing the history store cursor until the data store cursor is present. The increase in memory can be manageable.